R/deprecated.R

```diff
@@ -139,3 +153,92 @@ page_fill <- function(...) {
   .Deprecated("page_fillable", old = "page_fill")
   page_fillable(...)
 }
+
+#  Legacy Nav Containers --------------------------------------------------
+deprecate_if_not_called_from_shiny <- function(old_name, new, version) {
+  new_name <- deparse(substitute(new))
+
+  function(...) {
+    caller_fn_env <- environment(rlang::caller_fn())
+    if (!identical(rlang::env_name(caller_fn_env), "namespace:shiny")) {
+      msg <- sprintf(
+        "`%s()` was deprecated in {bslib} version %s, use `%s()` instead.",
+        old_name, version, new_name
+      )
+      .Deprecated(msg = msg)
+    }
+    new(...)
+  }
+}
+#' @rdname deprecated
+#' @aliases NULL
```

Comment:
Is this here since you want `?nav` to point to `?nav_panel`?

Reply:
Just a thought, we could also change `#' @rdname nav` to `#' @rdname nav-items`...

Reply:
And I wonder if we should redirect fhttps://rstudio.github.io/bslib/reference/nav.html?

Reply:
Ah yea, and if it helps, you can also drop stuff https://github.com/rstudio/shiny/blob/62bb21d5b6eb34a5701f849c78a04ff5efd5b46d/tools/documentation/pkgdown.yml#L221-L224